06-building-decays.md

```diff
@@ -0,0 +1,161 @@
+
+> ## Selections {.callout}
+> The `Selection` is the basic unit of the framework. It uses other `Selections` to process `LHCb::Particles` and writes them to a TES location easily findable through its `outputLocation` method. Additionally, it knows about other `Selections` that it requires to pass in order to obtain input particles through its `RequiredSelections` argument.
+> A `Selection` requires *all* its `RequiredSelections` to pass.
+>
+> There are several types of selections, namely
+>
+> - `Selection`, the most basic class.
+> - `MergedSelection`, which is used to join the output of several `Selection` objects in a single TES location.
+> - `AutomaticData` (also known as `DataOnDemand`), which builds objects from their TES location using a preconfigured map of (location, algorithm) pairs.
+
+> ## Selection sequences {.callout}
+> The `SelectionSequence` takes a `Selection` object, resolves its `Selection` requirements, and builds a flat, chained and ordered list of `Selections`. It then exports (via the `selection` method) a self-contained `GaudiSequencer` with all the algorithm configurables necessary to run the selection.
+> It also makes the output locations of the data written by the head of the selection chain available via the `outputLocations` method.
+
+To get our input particles we load the particle creation algorithms from the `CommonParticles` package:
+
+```python
+from CommonParticles.StdAllNoPIDsPions import StdAllNoPIDsPions as Pions
+from CommonParticles.StdAllLooseKaons import StdAllLooseKaons as Kaons
+```
+
+This package, which you can find [here](https://svnweb.cern.ch/trac/lhcb/browser/Stripping/trunk/Phys/CommonParticles/python/CommonParticles), allows to access premade particles with reasonable selections and gives easy access to common particles used in selections.
+
+> ## Using the `DataOnDemand` selection {.callout}
+> As discussed previously, the `DataOnDemand` or `AutomaticData` selection builds objects from their TES location.
+> In Gaudi, the TES location of the output of an algorithm is generally determined as `Phys/ALGO_NAME/OBJECT_TYPE`, where `OBJECT_TYPE` refers to `Particles`, `Vertices`, etc.
+> For example, in our specific case, we could have used the `DataOnDemand` class with the `Phys/StdAllNoPIDsPions/Particles` and `Phys/StdAllLooseKaons/Particles` as inputs for pions and kaons, respectively:
+> 
+> ```python
+> from PhysSelPython.Wrappers import DataOnDemand
+> Pions = DataOnDemand('Phys/StdAllNoPIDsPions/Particles')
+> Kaons = DataOnDemand('Phys/StdAllLooseKaons/Particles')
+> ```
+>
+> In fact, if you access the [code](https://svnweb.cern.ch/trac/lhcb/browser/Stripping/trunk/Phys/CommonParticles/python/CommonParticles/StdAllNoPIDsPions.py) for `StdAllNoPIDsPions`, you can see that in it the algorithm for creating the particles is instantiated and the map of the `DataOnDemand` service is updated via the `updateDoD` function.
```

Comment:
I meant `DataOnDemand` vs. `AutomaticData`; you can motivate using the former over the latter because it's used in the common particles (but again it's still arbitrary, so…).